Lesson16/aysel/aysel.js

```diff
@@ -0,0 +1,207 @@
+function isLeapYear(year){
+    if(year % 4 === 0 && year % 100 !== 0 || year % 400 === 0){
+        console.log('It\'s a leap year');
+    }
+    else{
+        console.log('It\'s not a leap year');
+    }
+
+}
+isLeapYear(4000);
+isLeapYear(2024);
+isLeapYear(1881);
+isLeapYear(3500);
+
+// 7. Write a function that takes in a temperature and checks if it's above or below freezing using if else statement.
+function checkTemperature (temp){
+    if(temp>0){
+        console.log('It\'s above freezing temperature');
+    }
+    else if(temp<0){
+        console.log('It\'s below freezing temperature');
+    }
+    else{
+        console.log('It\'s  freezing temperature');
+    }
+}
+
+checkTemperature(0)
+checkTemperature(-15)
+checkTemperature(6)
+
+
+// 8. Write a function that takes in a person's height and weight and checks if they are considered underweight, normal, overweight, or obese using if else statement.
+// Body Mass Index (BMI) is a person’s weight in kilograms (or pounds) divided by the square of height in meters (or feet). A high BMI can indicate high body fatness.
+//Formula: weight (kg) / [height (m)]2
+// Below 18.5	Underweight
+// 18.5 – 24.9	Healthy Weight
+// 25.0 – 29.9	Overweight
+// 30.0 and Above	Obesity
+
+function checkBMI(weight, height) {
+    
+    let bmi = weight / (height * height);
+
+  
+    if (bmi < 18.5) {
+        return 'Underweight';
+    } else if (bmi >= 18.5 && bmi <= 24.9) {
+        return 'Healthy Weight';
+    } else if (bmi >= 25 && bmi <= 29.9) {
+        return 'Overweight';
+    } else {
+        return 'Obesity';
+    }
+}
+
+
+console.log(checkBMI(70,1.72));
+console.log(checkBMI(190,1.82));
+console.log(checkBMI(40,1.72));
+
+
+// 9. Write a function that takes in a person's age and checks if they are a child (age 0-12), teenager (age 13-19), adult (age 20-64), or senior citizen (age 65 and above) using if else statement.
+function checkAges(age){
+    if(age>=0 && age<=12){
+        console.log('You are a child');
+    }
+    else if(age>=13 && age<=19){
+        console.log('You are a teenager');
+    }
+    else if(age>=20 && age<=64){
+        console.log('You are an adult');
+    }
+    else{
+        console.log('You are a senior citizen');
+    }
+}
+checkAges(3);
+checkAges(13);
+checkAges(23);
+checkAges(73);
+// 10. Write a function that takes age as a number and whether they are a student (yes or no). If the age is less than 18 and the user is a student, print "You are a student and not old enough to vote". If the age is between 18 and 65 (inclusive) and the user is not a student, print "You are eligible to vote". If the age is greater than 65, print "You are a senior citizen". If the user is a student but their age is greater than or equal to 18, print "You are old enough to vote".
+function checkVoterEligibility(age, isStudent) {
```

Comment:
For values like 'yes'/'no', in programming it's common to use Boolean type value. So change function body like this:

```

    if (age < 18 && isStudent) {
        console.log("You are a student and not old enough to vote");
    } else if (age >= 18 && age <= 65 && !isStudent) {
        console.log("You are eligible to vote");
    } else if (age > 65) {
        console.log("You are a senior citizen");
    } else if (isStudent && age >= 18) {
        console.log("You are old enough to vote");
    }
```

Reply:
See more info about booleans here: https://www.classes.cs.uchicago.edu/archive/2021/spring/11111-1/happycoding/p5js/if-statements.html#:~:text=An%20if%20statement%20checks%20a,when%20that%20value%20is%20true%20.

When calling the function, you need to call it like this: `checkVoterEligibility(17, true);`